service/src/test/java/com/ge/predix/acs/service/policy/evaluation/PolicyEvaluationWithAttributeReaderTest.java

```diff
@@ -83,35 +90,47 @@ public void setupMethod() throws Exception {
+        when(this.attributeReaderFactory.getResourceAttributeReader()).thenReturn(this.externalResourceAttributeReader);
+        when(this.attributeReaderFactory.getSubjectAttributeReader()).thenReturn(this.externalSubjectAttributeReader);
         PolicySet policySet = new ObjectMapper().readValue(
                 new File("src/test/resources/policy-set-with-one-policy-one-condition-using-res-attributes.json"),
                 PolicySet.class);
-        when(this.policyService.getAllPolicySets()).thenReturn(Arrays.asList(policySet));
+        when(this.policyService.getAllPolicySets()).thenReturn(Collections.singletonList(policySet));
+    }
 
+    private PolicyEvaluationResult evaluatePolicy() {
+        return this.evaluationService.evalPolicy(createRequest(RESOURCE_IDENTIFIER, SUBJECT_IDENTIFIER, ACTION));
+    }
+
+    @Test
+    public void testPolicyEvaluation() throws Exception {
         Set<Attribute> resourceAttributes = new HashSet<>();
         resourceAttributes.add(new Attribute("https://acs.attributes.int", "location", "sanramon"));
         resourceAttributes.add(new Attribute("https://acs.attributes.int", "role_required", "admin"));
-        BaseResource testResource = new BaseResource("/sites/1234", resourceAttributes);
+        BaseResource testResource = new BaseResource(RESOURCE_IDENTIFIER, resourceAttributes);
 
         Set<Attribute> subjectAttributes = new HashSet<>();
         subjectAttributes.add(new Attribute("https://acs.attributes.int", "role", "admin"));
-        BaseSubject testSubject = new BaseSubject("test-subject", subjectAttributes);
+        BaseSubject testSubject = new BaseSubject(SUBJECT_IDENTIFIER, subjectAttributes);
 
-        when(this.attributeReaderFactory.getResourceAttributeReader()).thenReturn(this.externalResourceAttributeReader);
         when(this.externalResourceAttributeReader.getAttributes(anyString())).thenReturn(testResource.getAttributes());
-        when(this.attributeReaderFactory.getSubjectAttributeReader()).thenReturn(this.externalSubjectAttributeReader);
         when(this.externalSubjectAttributeReader.getAttributesByScope(anyString(), anySetOf(Attribute.class)))
                 .thenReturn(testSubject.getAttributes());
 
-        PolicyEvaluationResult evalResult = this.evaluationService.evalPolicy(
-                createRequest(testResource.getResourceIdentifier(), testSubject.getSubjectIdentifier(), "GET"));
+        PolicyEvaluationResult evalResult = this.evaluatePolicy();
         Assert.assertEquals(evalResult.getEffect(), Effect.PERMIT);
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testPolicyEvaluationWhenAdaptersTimeOut() throws Exception {
+        when(this.externalResourceAttributeReader.getAttributes(Mockito.anyString()))
+                .thenThrow(AttributeRetrievalException.class);
+
+        PolicyEvaluationResult evalResult = this.evaluatePolicy();
+        Assert.assertEquals(evalResult.getEffect(), Effect.INDETERMINATE);
```

Comment:
this test would be better if you do a successful PERMIT first, then only mock the rule above to produce a INDETERMIANTE.   

Also, in this case eval service should set a appropriate message in PoliceEvaluationResult.message indicating some problem with the adapter, along with reference info for the adapter.  Just enough info, so the end-user can pass on the reference to service owner who is using ACS.